plugins/keymanagerng/app/javascript/widgets/app/components/secrets/secretDetails.jsx

```diff
@@ -88,18 +89,44 @@ const SecretDetails = () => {
       secret?.data?.content_types?.default,
     ],
     queryFn: getSecretPayload,
-    enabled: !!payloadRequested,
+    enabled:
+      payloadRequested || // Enable when download requested
+      (![
```

Comment:
I find this boolean expression a bit hard to read, it's easy to miss the `&& show` at the end. My suggestion:
```javascript
payloadRequested  || (show && ![....].includes(...))
```

Reply:
One additional suggestion. You use
```javascript
[
                      "application/octet-stream",
                      "application/pkcs8",
                      "application/pkix-cert",
                    ].includes(secret?.data?.content_types?.default) 
```
three times here. I think it would be much more readable if you wrote a little helper function `const isDownloadableDataType`

Then you can do e.g. 
```javascript
payloadRequested  || (show && !isDownloadableDataType(secret?.data?.content_types?.default))
```

Reply:
Fixed